### PR TITLE
Palindrome 270523

### PR DESCRIPTION
**Pull request by admiralrohan**

### CODE REVIEW

**Comment by taha-crafter** [comment]
Hi @admiralrohan 

Nice work here as well. Rohan👍🏼

The reason you have your solution in one commit is similar to the reason why you didn't reach a solution to the [FizzBuzz ](https://github.com/admiralrohan/the-software-essentialist/pull/6) assignment using "only" the examples from the grading checklist.

According to Khalil's advice, I have to add more tests (failure and success cases) for the same behavior... this way you will be forced to make changes to your production code early on.

So after your test for '`mom`' you could've added a failing test for '`bill`', instead of a test for '`Mom`' in b87873c16cbef8b699ecac60e369f32c81af7924  (notice that these tests, '`mom`' & '`bill`' [one success, and one failure test] are targeting the same behavior of _`"recognizing palindromes of single, same-case-letter words"`_ ) 

if you added a test for '`bill`' right after the test for '`mom`' then your test would fail early and you'd be forced to change the production code with a new fake it value (moving your solution forward)... if you don't add a failing test you'd still be in green for many tests as long as these tests are expected to return true... because returning `true` became the default behavior of your production code in the first green commit 4e0405730349d675ec0f73866e56eef2ea908c8a

In conclusion, grading checklist examples are not there to be followed exactly as is, instead we need to identify the high-level behaviors of the system and add success/failure example tests for each behavior, once we have a working solution we make sure that the grading checklist examples still pass to make sure our solution works as required.

Hope this helps.
